src/js/node/events.js

```diff
@@ -132,7 +138,9 @@ EventEmitterPrototype.addListener = function addListener(type, fn) {
     events[type] = [fn];
     this._eventsCount++;
   } else {
-    handlers.push(fn);
+    // for performance reasons the handlers are stored in reverse
+    // -> adding = unshifting
+    handlers.unshift(fn);
```

Comment:
`unshift` is slow... Look like either forward or backward there is a `unshift` use.